### PR TITLE
Pill selector filtering

### PR DESCRIPTION
**Pull request by danthecodingman**
Save the query that user filtered and clear. Selection of pills

### CODE REVIEW

File: fbuapp/Models/Query.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface Query : NSObject
+
+@property (nonatomic, strong) NSString *name;
+@property (nonatomic, strong) NSIndexSet *role;
+@property (nonatomic, strong) NSIndexSet *level;
+@property (nonatomic, assign) NSInteger category;
+@property (nonatomic, assign) float price;
```
Comment (belchercd) [inline fbuapp/Models/Query.h:19]: Do these need an asterisk?

File: fbuapp/View Controllers/FilterViewController.m [hunk 2]
```diff
@@ -22,57 +23,109 @@ @interface FilterViewController () <CategoryHeaderViewDelegate>
...
+    [self.pillSelector mas_makeConstraints:^(MASConstraintMaker *make) {
         make.edges.equalTo(self.eventCategoryView).with.insets(UIEdgeInsetsMake(0, 0, 0, 0));
     }];
+
+    [self initializeWithSavedQuery];
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary  *)change context:(void *)context
+{
+    [self.pillSelector.collectionView selectItemAtIndexPath : [NSIndexPath indexPathForItem:self.eventCategory inSection:0] animated: NO scrollPosition: UICollectionViewScrollPositionNone];//dka
```
Comment (belchercd) [inline fbuapp/View Controllers/FilterViewController.m]: What is this comment?

File: fbuapp/View Controllers/HomeViewController.m [hunk 3]
```diff
@@ -20,22 +20,28 @@
...
 
-- (void)viewDidLoad {
+- (void)viewDidLoad{
     [super viewDidLoad];
     self.feed.numberOfPosts = 0;
     self.postQuery = [Post query];
-    self.feed = [[PostTableView alloc] initWithUserId:@"myuserid"];
+    self.savedQuery = [[Query alloc] init];
+    
+    self.feed = [[PostTableView alloc] initWithUserId:@"myuserid"];//[try current user here]
```
Comment (belchercd) [inline fbuapp/View Controllers/HomeViewController.m:45]: Does this need to be set to the current user's objectId?

File: fbuapp/View Controllers/HomeViewController.m [hunk 4]
```diff
@@ -44,21 +50,22 @@ - (void)viewDidLoad {
...
+- (IBAction)presentFilterViewController:(id)sender{
     FilterViewController *filterVCObj =[[FilterViewController alloc] initWithNibName:@"FilterViewController" bundle:nil];
     filterVCObj.currentLocation = [PFGeoPoint geoPointWithLocation: self.feed.currentLocation];
+    filterVCObj.savedQuery = self.savedQuery;
     [self presentViewController:[[UINavigationController alloc] initWithRootViewController:filterVCObj] animated:YES completion:nil];
     filterVCObj.delegate = self;
 }
 
-- (void) fetchPosts {
+- (void)fetchPosts{
```
Comment (belchercd) [inline fbuapp/View Controllers/HomeViewController.m:69]: Not sure if it's just my view, but there seems to be a chunk of info missing. Such as extra starter brackets and an if statement.

File: fbuapp/Views/PostTableView.m [hunk 5]
```diff
@@ -97,12 +97,18 @@ -(void)locationManager:(CLLocationManager *)manager didFailWithError:(NSError *)
...
 - (UITableViewCell *)tableView:(UITableView *)tableView
          cellForRowAtIndexPath:(NSIndexPath *)indexPath
 {
     PostCell *cell = [tableView dequeueReusableCellWithIdentifier:kTableViewPostCell];
     cell.delegate = self;
     Post *post = self.posts[indexPath.row];
+    
+
+    
+//    cell.eventAuthor.text = [PFQuery getUserObjectWithId: post[@"eventAuthor"]][@"firstName"];
```
Comment (eazheng) [inline fbuapp/Views/PostTableView.m]: probably delete this when merging with camryn's branch.

File: fbuapp/Views/CategoryHeaderView.m [hunk 6]
```diff
@@ -89,14 +89,24 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cell
     cell.eventCategory.text = category.name;
     cell.pillBackground.layer.cornerRadius = cell.pillBackground.frame.size.height / 2;
     
+    
+    
     return cell;
 }
 
 - (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath {
+    [collectionView selectItemAtIndexPath:indexPath animated:NO scrollPosition:UICollectionViewScrollPositionNone];//dka not nec
```
Comment (eazheng) [inline fbuapp/Views/CategoryHeaderView.m]: what's dka

File: fbuapp/View Controllers/HomeViewController.m (hunk 4 above)
Comment (eazheng) [reply fbuapp/View Controllers/HomeViewController.m:69]: oh yeah i think it skipped lines 72 - 79

File: fbuapp/View Controllers/HomeViewController.m [hunk 7]
```diff
@@ -93,19 +105,23 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView {
...
 
-- (void) filterPostsWithQuery: (PFQuery *) postQuery{
+- (void)filterPostsWithQuery:(PFQuery *)postQuery withSavedQuery:(Query *)savedQuery{
     self.postQuery = postQuery;
+    [self.pillSelector.collectionView deselectItemAtIndexPath: [NSIndexPath indexPathForItem: self.savedQuery.category inSection:0] animated:NO];
+    self.savedQuery = savedQuery;
+    [self.pillSelector.collectionView selectItemAtIndexPath: [NSIndexPath indexPathForItem: self.savedQuery.category inSection:0] animated:NO scrollPosition:UICollectionViewScrollPositionNone];
     self.feed.numberOfPosts = 0;
     [self fetchPosts];
+    [self.feed setContentOffset:CGPointMake(0,-62)];
```
Comment (eazheng) [inline fbuapp/View Controllers/HomeViewController.m:116]: where does the number -62 come from

File: fbuapp/View Controllers/HomeViewController.m (hunk 3 above)
Comment (danthecodingman) [reply fbuapp/View Controllers/HomeViewController.m:45]: Yes this should but I never had current user working on my machine so it hasn't been tested yet

File: fbuapp/View Controllers/HomeViewController.m (hunk 7 above)
Comment (danthecodingman) [reply fbuapp/View Controllers/HomeViewController.m:116]: The real height of the pill selector.

File: fbuapp/Views/PostTableView.m (hunk 5 above)
Comment (danthecodingman) [reply fbuapp/Views/PostTableView.m]: ok

**Pull request merged**